### PR TITLE
Include full alignment and APE for Pixel Phase-I detector

### PR DESCRIPTION
**Pull request by mmusich**
# Summary of changes in Global Tags
## Upgrade
- **PhaseI realistic scenario** : [80X_upgrade2017_realistic_v5](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/80X_upgrade2017_realistic_v5) as [80X_upgrade2017_realistic_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/80X_upgrade2017_realistic_v4) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/80X_upgrade2017_realistic_v5/80X_upgrade2017_realistic_v4): 
  - Ideal Tracker Alignment in Phase-I pixel, realistic run2 MC in Strips  
  - Ideal Tracker Alignment Errors in Phase-I pixel, realistic run2 MC in Strips
  - Ideal surface deformations in Phase-I pixel (null) and realistic Run2 surface deformations in Strips
- **PhaseI design scenario** : [80X_upgrade2017_design_v13](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/80X_upgrade2017_design_v13) as [80X_upgrade2017_design_v12](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/80X_upgrade2017_design_v12) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/80X_upgrade2017_design_v13/80X_upgrade2017_design_v12): 
  - Ideal Tracker Alignment in Phase-I pixel
  - Ideal Tracker Alignment Errors in Phase-I pixel

This should allow to switch on the `applyAlignment` parameter of `trackerGeometry`

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by mmusich** [comment]
attn: @makortel

File: Configuration/Geometry/python/GeometryExtended2017Reco_cff.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 from Geometry.TrackerGeometryBuilder.idealForDigiTrackerGeometry_cff import *
 from Geometry.CSCGeometryBuilder.idealForDigiCscGeometry_cff import *
 from Geometry.DTGeometryBuilder.idealForDigiDtGeometry_cff import *
-trackerGeometry.applyAlignment = cms.bool(False)
+trackerGeometry.applyAlignment = cms.bool(True)
```
Comment (makortel) [inline Configuration/Geometry/python/GeometryExtended2017Reco_cff.py]: May I suggest to just remove this override instead (I think it would be more clear)? The default is `True` https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Geometry/TrackerGeometryBuilder/plugins/TrackerDigiGeometryESModule.cc#L64.

**Comment by makortel** [comment]
Thanks @mmusich!

Sorry to hijack the PR a bit, but on a related note could somebody clarify if the 2017 RECO workflow currently reads the geometry from XML or from GT (RECO step loads "Configuration.Geometry.GeometryExtended2017Reco_cff"; pardon my ignorance on the details of geometry technicalities)? The geometry is in the GTs anyway, right?

@boudoul @ianna

**Comment by mmusich** [comment]
@makortel that's my understanding (i.e. geometry taken from GT anyway). Please @ianna confirm

**Comment by mmusich** [comment]
```
$ conddb list 80X_upgrade2017_design_Queue | grep GeometryFileRcd
GeometryFileRcd                                         Extended                                         XMLFILE_Geometry_2017_80YV2_Extended2017_mc                                     
GeometryFileRcd                                         Ideal                                            XMLFILE_Geometry_Run2_76YV2_Ideal2015_mc        
                                       
$ conddb list 80X_upgrade2017_realistic_Queue | grep GeometryFileRcd
GeometryFileRcd                                         Extended                                         XMLFILE_Geometry_2017_80YV2_Extended2017_mc                                          
GeometryFileRcd                                         Extended2015CastorMeasured                       XMLFILE_Geometry_80YV1_Extended2015CastorMeasured_mc                                 
GeometryFileRcd                                         Extended2015CastorSystMinus                      XMLFILE_Geometry_80YV1_Extended2015CastorSystMinus_mc                                
GeometryFileRcd                                         Extended2015CastorSystPlus                       XMLFILE_Geometry_80YV1_Extended2015CastorSystPlus_mc                                 
GeometryFileRcd                                         Extended2015ZeroMaterial                         XMLFILE_Geometry_Run2_76YV2_Extended2015ZeroMaterial_mc                              
GeometryFileRcd                                         Extended2015dev                                  XMLFILE_Geometry_Run2_76YV2_Extended2015dev_mc                                       
GeometryFileRcd                                         Ideal                                            XMLFILE_Geometry_Run2_76YV2_Ideal2015_mc                                             
GeometryFileRcd                                         Ideal2015dev                                     XMLFILE_Geometry_Run2_76YV2_Ideal2015dev_mc                                          
```
also: all the the other geometry labels in GT other than `extended` are run2 related. I would just remove them in the next iteration. Is it fine with you @ianna?

**Comment by mmusich** [comment]
please test

File: Configuration/Geometry/python/GeometryExtended2017Reco_cff.py (hunk 1 above)
Comment (mmusich) [inline Configuration/Geometry/python/GeometryExtended2017Reco_cff.py]: @makortel done in the latest commit

**Comment by ianna** [comment]
@makortel - yes, geometry is taken from GT, but the alignment was not applied.

**Comment by ianna** [comment]
+1

**Comment by makortel** [comment]
Thanks @ianna. For my education, why do we need `GeometryExtended2017Reco_cff` for 2017 (reco step) when 2016 uses 'GeometryRecoDB_cff`? Was it just to disable alignment, or is there something more?

**Comment by boudoul** [comment]
hi @ianna , all - In relvals the geometry is set up (in runthematrix ) with --geometry Extended2017 , which leads to the cff taht Matti was reminding , so I would naively think that we are using xml instead of GT fro the geometry

**Comment by mmusich** [comment]
+1

**Comment by makortel** [comment]
I have the tracking validation plots with 1000 TTbar+35PU events here
https://mkortela.web.cern.ch/mkortela/tracking/validation/CMSSW_8_1_0_pre5_pr14545/
Changes are negligible.

**Comment by mmusich** [comment]
@makortel thanks, can you clarify which wf is this, or better said which set of conditions are you comparing here?

**Comment by makortel** [comment]
@mmusich Blue is 80X_upgrade2017_realistic_v4 and red is 80X_upgrade2017_realistic_v5.

**Comment by makortel** [comment]
Workflow is essentially 10024.0 (but I use TTbar+PU DIGI-RAW from 810pre4 as an input).

**Comment by mmusich** [comment]
@markotel, basically what changed between the two is Strips to Ideal alignment to Strips to very good alignment (asymptotic). All in all I would say is expected to see small differences.

**Comment by davidlange6** [comment]
+1
however contrary to the above, given this:
--geometry Extended2017
in the cmsDriver commands currently, I would expect the xml geometry rather than the DB one to be used in the current IBs.

**Comment by makortel** [comment]
(there is probably a better forum to continue the discussion, but anyway)

Is there any reason to not to switch 2017 workflows to take geometry from GT (now that this PR is merged)?

I guess switching to GT would give some flexibility on when exactly to enable the upcoming updates on the phase1 pixel geometry. I'm thinking specifically release validation, where I'd much prefer to disentangle significant geometry changes from other changes (code/configuration/other conditions). By significant I mean something having effect larger than the statistical fluctuations from DIGI regeneration (so probably BPix finetuning is not significant, but making FPix realistic could be). Maybe geometry from GT would play more nicely together with alignment as well?

**Comment by mmusich** [comment]
@makortel I would be in favor of your proposal. The alignment in the pixel is currently "ideal" meaning is just a copy in another format of the reco geometry. Updating together the geometry and the alignment from GT, rather than chasing the XMLs, should make life much easier.

**Comment by makortel** [comment]
I tried blatantly to run 10024.0 step3 in CMSSW_8_1_X_2016-05-19-2300 by dropping the `--geometry Extended2017` (as in e.g. 1325.0, so that the resulting job configuration loads "Configuration.StandardSequences.GeometryRecoDB_cff"), but I get an exception
```
----- Begin Fatal Exception 20-May-2016 14:29:19 CEST-----------------------
An exception of category 'Conditions mismatch' occurred while
   [0] Processing run: 1
   [1] Running path 'dqmoffline_step'
   [2] Calling beginRun for module DigiTask/'digiTask'
Exception Message:
Requested conditions of type HcalChannelQuality for cell (0x43180801) (HB 2,1,1) got conditions for cell (0x42006089) (HB 1,9,1)
----- End Fatal Exception -------------------------------------------------
```
so apparently the XML-side is still needed (or there is some other customization/hardwiring done somewhere).

If we would take the "geometry from GT" as a goal, what should be done to get there? @ianna @mmusich @boudoul

**Comment by mmusich** [comment]
@abdoulline, can you remind us when this issue in https://github.com/cms-sw/cmssw/pull/14545#issuecomment-220594312 is supposed to happen?
Looks like a mismatch in the detIDs used in the Hcal conditions w.r.t. what is used in the geometry. Should we have all of them regenerated accordingly?

**Comment by abdoulline** [comment]
I'm afraid I'm out of context here...
For 2016, please try
process.load("Configuration.Geometry.GeometryExtended2016Reco_cff")

**Comment by makortel** [comment]
@abdoulline Context is 2017 and an attempt to take the geometry via GlobalTag instead of "Configuration.Geometry.GeometryExtended2017Reco_cff".

**Comment by abdoulline** [comment]
Ah... OK. 
There is no Offline HCAL DB conditions for 2017 yet. Only hardcoded ones.
Remaking 2017 conditions (completely different number of channels, DetId space) is in the HCAL agenda.

**Comment by mmusich** [comment]
@makortel there you go...

**Comment by makortel** [comment]
Ok, thanks @abdoulline.

Would it be possible to take tracker (well, everything else than HCAL) from GT and HCAL via hardcoding, or would this be too hacky and not worth of the effort?

**Comment by abdoulline** [comment]
I take the liberty to include Sunanda (as HCAL geometry expert)  @bsunanda  in the discussion.

**Comment by makortel** [comment]
To not let the discussion dry (I'd like to get an explicit answer either way), can anybody tell if we can combine tracker geometry (or everything else than HCAL) from GT and HCAL geometry from hardcoding?

Or should we just accept the current situation? I.e. needing to update the alignment every time phase1 pixel geometry is updated (IIUC), and making it more difficult to isolate effects by geometry changes from other changes (mainly concerning release validation; although the current plan to freeze tracking for a while after #14621 and PR-to-be-submitted-after-#14544 helps that).

@boudoul @mmusich @bsunanda

**Comment by ghellwig** [comment]
> Or should we just accept the current situation? I.e. needing to update the alignment every time phase1 pixel geometry is updated (IIUC)

@makortel just to avoid confusion: if the geometry changes, the TrackerAlignmentRcd has to be updated anyway because it does not contain relative differences, but the absolute positions and rotations of the modules.

**Comment by bsunanda** [comment]
Hcal geometry can be in DB but the conditions are not there in as yet. So in principle one can run with Hcal but conditions will be hardcoded numbers. There are efforts underway to put Hcal conditions to DB.
________________________________

**Comment by mmusich** [comment]
@makortel IMHO what matters for you is more the alignment than the reco geometry itself that at this point is overridden. The important thing is to keep the simulation geometry and the alignment in synch at this point. Wouldn't this be enough?

**Comment by makortel** [comment]
@mmusich I agree that keeping alignment and simulation geometry in sync should be enough. But this requires some effort, right? (but maybe not much?)

My main concern has been the release validation, where I'd like to be able disentangle the (significant) geometry updates from other changes (so mainly FPix I guess). My understanding is that with GT-based geometry this would be easy (two sets of RelVals per 2017 pre-release: one to compare previous release, and other to compare geometry update), but with XML geometry it would be more difficult. If we manage to freeze the tracking in 810pre6 (I have one PR more to be submitted once #14544 gets merged), this point is probably not as big of an issue what I feared.

In summary, if it is not currently easy to get the tracker geometry from GT, I can probably live with the gradual updates on the (pixel) XML geometry.

**Comment by ianna** [comment]
@makortel - AFAIU, a GT is designed to be able keep frequent alignment updates. I'd also think, an ideal Pixel geometry should not be changing much by now. When you create a simulation geometry, you take the latter and apply the first. Unless you record this though GT (by uploading appropriate conditions), reproducing the same result is not guaranteed. I think the only way to trust any release validation is to go though GT. It does require (a little) effort and some discipline: update xml, test it, upload it to Prep DB, test it, upload it to Prod DB, test it, request GT update, test it ;-)

**Comment by makortel** [comment]
@ianna My understanding of the status of the phase1 pixel geometry is a bit different (based on e.g.  Tracker DPG-Tracking POG meeting on May 13 https://indico.cern.ch/event/512685/contributions/2167875/attachments/1273212/1887755/13052016_MatBudgNews.pdf), that is, FPix needs major changes. Or have these been integrated already during the past two weeks and I've just missed them? @boudoul @veszpv could you please clarify?

**Comment by veszpv** [comment]
@makortel No, you are right, we are still planning a major change for the FPix geometry. This has got stuck a bit somewhere before validating the new geometry, therefore no PR has been made, yet. I will try to push this a bit to happen in the next couple of weeks. But we should clearly converge with the question on configuring from xml or DB so that we have a clear procedure by then.

**Comment by mmusich** [comment]
@veszpv  
> But we should clearly converge with the question on configuring from xml or DB so that we have a clear procedure by then.

IMHO you need to provide both, possibly with a version without issues in the material description to let the tracker alignment workflow move along. Switching to take the full CMS geometry from DB is unfortunately coupled to the conditions of other subdetectors (namely Hcal), so depends mostly on their schedule.

**Comment by ianna** [comment]
@makortel, @mdhildreth, @civanch, @davidlange6, @abdoulline @bsunanda, @boudoul, @kpedro88  - Having spent some time debugging the issue with Hcal conditions here is a proposed solution for 2017 workflows. It works locally for me. Please, let me know if you'd prefer me to submit it as a PR or someone else is going to take care of it.

1. Allow default 2017 DB geometry usage from 2017 GT by removing 2017 label from upgradeGeoms: 
http://cmslxr.fnal.gov/source/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py?v=CMSSW_8_1_X_2016-05-31-2300#0015

2. Modify/add 2017 HcalHardcodeCalibrations producer to request (a minimum required to run without throwing):
```
process.es_hardcode.toGet = cms.untracked.vstring('GainWidths', 
    'QIETypes', 
    'Gains', 
    'MCParams', 
    'PedestalWidths', 
    'Pedestals', 
    'RespCorrs', 
    'TimeCorrs', 
    'LUTCorrs')
process.es_hardcode.GainWidthsForTrigPrims = cms.bool(True)
```
Note, it is possible to ask for "all" conditions rather then one by one.

**Comment by kpedro88** [comment]
@ianna - I am a little confused about the problem here. The default 2017 geometry has the Run1 HCAL everywhere, except for 1x1 TPs in HF. Cloning the current 2016 HCAL MC conditions should work for the 2017 geometry.

For the 2017*dev* geometry, where there are many new HCAL cells, DB conditions are not yet available. There are several "development customizations" defined for HCAL in https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/SLHCUpgradeSimulations/Configuration/python/HCalCustoms.py. The main purpose of these is to enable the hardcode conditions.

RECO for QIE10/QIE11 is not ready yet. With `customise_Hcal2017`, everything will run, but RecHits will be missing for HF. With `customise_Hcal2017Full`, RecHits will be missing for HE and HF. I could define one like `customise_Hcal2017Old`, which would produce RecHits for both HE and HF in the new geometry, but I'm not sure if there's a need for this at the moment.

**Comment by ianna** [comment]
@kpedro88 - Indeed, because HCAL DB conditions are not yet available, mixing module request for the conditions should be passed to HcalHardcodeCalibrations producer. The exception thrown is at a DIGI step. If a 2017 HCAL customization is already there, why not use it?

**Comment by kpedro88** [comment]
@ianna - can you clarify which version of the geometry is causing a problem? If it is just Extended2017, you should just clone the existing 2016 HCAL DB conditions for the GT. If it is Extended2017dev, we have to use some version of the hardcode conditions, which can be picked up from one of the existing conditions, or I can make `customise_Hcal2017Old` if we want to get RecHits for every subdetector.

**Comment by ianna** [comment]
@kpedro88 - This happens with a default Extended geometry from 2017 GT which is equivalent to Extended2017 built from xml files. The tag is:
``` 
XMLFILE_Geometry_2017_80YV2_Extended2017_mc
```

**Comment by makortel** [comment]
Out of curiosity, what do we have for HCAL in  [80X_upgrade2017_realistic_v5](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/80X_upgrade2017_realistic_v5) GT? I see stuff like
```
GeometryFileRcd    Extended   XMLFILE_Geometry_2017_80YV2_Extended2017_mc
HcalParametersRcd     -       HCALParameters_Geometry_2017_80YV2
PHcalRcd              -       HCALRECO_Geometry_2017_80YV2
```
but I have no idea what exactly those mean.

**Comment by ianna** [comment]
@makortel - Here are some notes on what the tag versions mean:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideDataBaseGeometry#Payloads_In_the_Production_Datab

**Comment by kpedro88** [comment]
Also, out of general curiosity: is it possible to compare two GTs directly on the new conddb website? This feature was definitely available on the old site...

**Comment by mmusich** [comment]
@kpedro88 yes it is. Which GT do you want to compare?

**Comment by ianna** [comment]
@kpedro88 - yes, you select them, then diff. 
I'll check Extended2017 scenario and update it in DB accordingly.

**Comment by kpedro88** [comment]
Ah, I have to search for the tag to see the diff, I can't just enter the URL directly.

**Comment by mmusich** [comment]
@ianna @kpedro88 this looks like going back to me. Why is not possible to create new conditions compliant with the latest version of the geometry?

**Comment by mmusich** [comment]
@kpedro88  

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/<GT1>/<GT2> 

should fullfill your desires...

**Comment by kpedro88** [comment]
@mmusich - it is possible to create fake conditions (which may make questionable physical sense), but this is somewhat manpower-intensive at the moment for HCAL. Since a lot of things are still in development, so far we have deemed it easier just to use the hardcode conditions for our work. Pixel development doesn't need to know about the HCAL changes, so better to factorize in my opinion.

Thanks for the diff URL, much easier than using the browser...

**Comment by ianna** [comment]
@mmusich - I think, it is actually moving forward :-)

Hcal parameters have been updated recently for 2016, but 2017 scenario was not, though it should have been:
```
[yana@cmsdev03 src]$ diff Geometry/HcalCommonData/data/Run2/hcalSimNumbering.xml Geometry/HcalCommonData/data/Run2/hcalSimNumbering16.xml
59c59
<      2,  3,  2,  4
---
>      2,  3,  4,  4
68,69c68,71
<   <Vector name="noff" type="numeric" nEntries="18">
<     15, 29,  4, 10,  5,  2,  4, 18, 19, 11, 12, 13, 14,  2,  3,  2,  1,  0
---
>   <Vector name="noff" type="numeric" nEntries="42">
>     15, 29,  4, 10,  5,  2,  4, 18, 19, 11, 12, 13, 14,  2,  3,  2,  1, 12,
>     29, 39, 30, 39, 31, 39, 32, 39, 33, 39, 34, 39, 35, 39, 36, 39,
>     37, 39, 38, 39, 39, 39, 41, 39
87c89
<     2.840, 2.090, 0.000, 0.000
---
>     2.840, 2.090, 2.840, 2.090
[yana@cmsdev03 src]$ diff Geometry/HcalCommonData/data/Run2/hcalRecNumbering.xml Geometry/HcalCommonData/data/Run2/hcalRecNumbering16.xml
40,41c40,41
<     <Parameter name="TopologyMode" value="HcalTopologyMode::LHC" eval="false"/>
<     <Parameter name="TriggerMode"  value="HcalTopologyMode::tm_LHC_RCT" eval="false"/>
---
>     <Parameter name="TopologyMode" value="HcalTopologyMode::SLHC" eval="false"/>
>     <Parameter name="TriggerMode"  value="HcalTopologyMode::tm_LHC_RCT_and_1x1" eval="false"/>
```

**Comment by mmusich** [comment]
@mmusich, going back and forth with the geometry in GT is manpower-intensive for AlCa, especially considering the current proposal is to step back to: Hcal 2015 + 1x1 HF TPs.
PS: also `conddb diff` will give you the same output

**Comment by mmusich** [comment]
@ianna, is the diff you point out supposed to cure the inconsistencies spotted by @makortel in https://github.com/cms-sw/cmssw/pull/14545#issuecomment-220594312?

**Comment by kpedro88** [comment]
@mmusich - I thought it had been agreed for a long time not to put Extended2017dev in the global tag at all until it was ready. HCAL seems to get yelled at no matter what we do.

@ianna - the changes you point out are for Extended2016dev, with the few new HF QIE10 cells. This was also agreed not to go in the GT until conditions are available. In this case, they hopefully will be soon, but are not yet. @abdoulline may know more about the status for 2016dev.

**Comment by ianna** [comment]
@kpedro88 - It's comparing Extended2017 with Extended2016 (no dev)

**Comment by kpedro88** [comment]
@ianna - ah, this is because of weird historical naming conventions. Extended2016 is actually the dev version, while Extended2016a is the version currently used in 80X, with 2015 HCAL + 1x1 HF TPs. @bsunanda, please confirm.

**Comment by ianna** [comment]
@mmusich - yes, this is a fix https://github.com/cms-sw/cmssw/pull/14545#issuecomment-223003792

**Comment by ianna** [comment]
@kpedro88 - and Extended2017dev is not dev?

**Comment by kpedro88** [comment]
@ianna - let me fully clarify the naming situation.

* https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Configuration/Geometry/python/GeometryExtended2016_cff.py imports Geometry.CMSCommonData.cmsExtendedGeometry2016aXML_cfi
* https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Configuration/Geometry/python/GeometryExtended2016dev_cff.py imports Geometry.CMSCommonData.cmsExtendedGeometry2016XML_cfi
* https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Configuration/Geometry/python/GeometryExtended2017_cff.py imports Geometry.CMSCommonData.cmsExtendedGeometry2017XML_cfi
* https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Configuration/Geometry/python/GeometryExtended2017dev_cff.py imports  Geometry.CMSCommonData.cmsExtendedGeometry2017devXML_cfi

When I'm using the "dev" name, I'm thinking of the Configuration/Geometry configs that actually get loaded by the process. The XML config files are unfortunately not named consistently with the higher-level configs that use them. Neither of the "dev" configs should be in GTs at the moment.

**Comment by ianna** [comment]
@kpedro88 - so Extended2017 differs from Extended2016a:
```
[yana@cmsdev03 src]$ diff Geometry/HcalCommonData/data/Run2/hcalSimNumbering.xml Geometry/HcalCommonData/data/Run2/hcalSimNumbering16a.xml
87c87
<     2.840, 2.090, 0.000, 0.000
---
>     2.840, 2.090, 2.840, 2.090
[yana@cmsdev03 src]$ diff Geometry/HcalCommonData/data/Run2/hcalRecNumbering.xml Geometry/HcalCommonData/data/Run2/hcalRecNumbering16a.xml
41c41
<     <Parameter name="TriggerMode"  value="HcalTopologyMode::tm_LHC_RCT" eval="false"/>
---
>     <Parameter name="TriggerMode"  value="HcalTopologyMode::tm_LHC_RCT_and_1x1" eval="false"/>
```

**Comment by kpedro88** [comment]
@bsunanda will have to clarify that. I was not aware of that difference.

**Comment by ianna** [comment]
@kpedro88 - yep, we have agreed on naming conventions, but they were not respected for 2016 scenarios:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideDetectorDescription#Adding_New_Geometry_Scenario

**Comment by ianna** [comment]
@kpedro88 and @bsunanda - Hcal in Extended2017 scenario is identical to the one in 2015.  Plus new phase 1 pixel detector is added. BHM, BCM and PLT are removed. Ideally, current Hcal conditions should work.

**Comment by bsunanda** [comment]
Hi

Extended2017 is the same as Extended2015 for HCAL right now which is worse that Extended2016a and surely of the two dev ones.

Sunanda
________________________________

**Comment by mmusich** [comment]
All: this looks pretty much like a mess. @ianna would it be possible to create a geometry which is like Extended2016a for Hcal and Extended2017 for everything else?

**Comment by kpedro88** [comment]
@mmusich - I agree with both of your comments...

**Comment by ianna** [comment]
@mmusich - yes, I'll do that.

**Comment by abdoulline** [comment]
Wow... 
Sorry for not participating (I had an appointment)...

When I've participated today in a round-table discussion with Yana/David/Vladimir/Michael, I had the impression it was about 2017dev (means new HCAL in 2017, not 2017 with 2015 HCAL), so I've suggested GT Geometry + hardcoded conditions for  2017dev.
And also answered to Yana's question about hardcoded->DB 2017dev conditions: under way, but no deadlines.   BTW, we have had (successfully) hardcoded conditions in SLHC releases for 2019/2023,  while ECAL was very busy making an re-making all the time a number of conditions in DB for various lumis...

**Comment by kpedro88** [comment]
@abdoulline - I think in general, there is a place for hardcoded (read: algorithmically generated outside of DB) conditions, when dealing with rapidly changing conditions. Aging studies are a good example, and I think that HCAL upgrade development should be another example, as we won't know the real conditions until the hardware is ready. But in this immediate case, I think we might as well avoid complicating the phase 1 pixel development if we can.

**Comment by abdoulline** [comment]
Agree.

**Comment by mmusich** [comment]
Thanks all for the constructive feedback. Follow-up at https://github.com/cms-sw/cmssw/pull/14748
